core-bundle/contao/library/Contao/DcaLoader.php

```diff
@@ -164,7 +172,7 @@ private function loadDcaFiles()
 		$strCachePath = $strCacheDir . '/contao/dca/' . $this->strTable . '.php';
 
 		// Try to load from cache
-		if (file_exists($strCachePath) && !System::getContainer()->getParameter('kernel.debug'))
+		if (file_exists($strCachePath) && (!System::getContainer()->getParameter('kernel.debug') || isset(self::$freshPaths[$strCachePath])))
```

Comment:
Would make sense to swap the arguments of `&&` to short circuit the file exists check when in dev mode or cache miss, no?

Reply:
```suggestion
		if ((!System::getContainer()->getParameter('kernel.debug') || isset(self::$freshPaths[$strCachePath])) && file_exists($strCachePath))
```